File: Souqly-SPA/_models/Product.ts
import { Category } from "./Category";
import { Image } from "./Image";
import { MarketingProduct } from "./MarketingProduct";
import { Option } from "./Option";
import { ProductOptionCart } from "./ProductOptionCart";
import { User } from "./user";

export class Product {
    constructor(public Id?:number, public ProductName?:string, public Weight?:number, public Description?:string,
        public Brand?:string, public Date?:Date, public Dimension?:string, public Category?:Category,
        public CategoryId?:number, public Options?:Option[], public Images?:Image[],
            public ProductCarts?:ProductOptionCart[], public MarketingProducts?:MarketingProduct[],
            public Supplier?:User[], public SupplierId?:number){

    }
}

File: Souqly-API/Models/Product.cs
using System.Net.Mime;
using System;
using System.Collections.Generic;

namespace Souqly_API.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string ProductName { get; set; }
        public float Weight { get; set; }
        public string Description { get; set; }
        public string Brand { get; set; }
        public DateTime Date { get; set; }
        public string Dimension { get; set; }

        public User User { get; set; }
        public Category Category { get; set; }
        public int CategoryId { get; set; }
        public int UserId { get; set; }
 
        public ICollection<Option> Options { get; set; }
        public ICollection<Image> Images { get; set; }

        // Many To Many Realation
        public List<ProductOptionCart> ProductCarts { get; set; }
        public virtual ICollection<MarketingProduct> MarketingProducts { get; set; }


    }
}